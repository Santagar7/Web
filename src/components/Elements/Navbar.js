import { Link } from "react-router-dom";
import {useState,useEffect} from 'react';
import ActionsPanel from '../Elements/ActionsPanel';
import MembersPanel from '../Elements/MembersPanel';
import React from 'react';
function Navbar({index='', val=false,refresh=(()=>{})}){
    const [showActions,setShowActions] = useState(false);
    const [showMembers,setShowMembers] = useState(false);

    function LogOut(){
        console.log('log out');
    }

    function ShowActionsPanel(e){
        e.preventDefault();
        setShowActions(!showActions);
        setShowMembers(false);
    }
    function ShowMembersPanel(e){
        e.preventDefault();
        setShowMembers(!showMembers);
        setShowActions(false);
    }

    useEffect(()=>{
        refresh(!val);
    },[showActions,showMembers])

    return(
        <div className="extra">
            <div className="topnav">
                <a className="active" onClick={LogOut} >Log Out</a>
                <Link id="members" to={'#'} onClick={ShowMembersPanel}  data-testid='members'>Members</Link>
                null:<Link to="/budgets"  data-testid='budgets'>Budgets</Link>
                null:<Link id="action" to={'#'} data-testid='action' onClick={ShowActionsPanel}>Actions</Link>
            </div>
            {showActions?<ActionsPanel setActive={setShowActions}/>:null}
            {showMembers?<MembersPanel setActive={setShowMembers}/>:null}
        </div>
    )
// <ActionsPanel active={showActions} setActive={setShowActions}/>
}
export default Navbar;