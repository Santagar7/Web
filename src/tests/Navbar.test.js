import { BrowserRouter } from 'react-router-dom';
import { expect, test } from '@jest/globals';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Navbar from '../components/Elements/Navbar';

test('should render correctly', () => {
    const tree = renderer.create(<BrowserRouter><Navbar/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

test('check create new action', () => {
    render(
        <BrowserRouter>
            <Navbar />
        </BrowserRouter>,
    );

    const linkElement = screen.getByText(/Actions/i);
    expect(linkElement).toBeInTheDocument();
});

test('check actions panel', () => {
    render(
        <BrowserRouter>
            <Navbar />
        </BrowserRouter>,
    );
    const submit = screen.getByText(/Actions/i);
    fireEvent.click(submit);
    const linkElement = screen.getByText(/Create new Action/i);
    expect(linkElement).toBeInTheDocument();
});

test('check members panel', () => {
    render(
        <BrowserRouter>
            <Navbar />
        </BrowserRouter>,
    );
    const submit = screen.getByText(/Members/i);
    fireEvent.click(submit);
    const linkElement = screen.getByText(/All Members/i);
    expect(linkElement).toBeInTheDocument();
});