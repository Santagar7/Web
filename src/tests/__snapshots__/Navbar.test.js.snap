// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="extra"
>
  <div
    className="topnav"
  >
    <a
      className="active"
      onClick={[Function]}
    >
      Log Out
    </a>
    <a
      data-testid="members"
      href="/"
      id="members"
      onClick={[Function]}
    >
      Members
    </a>
    null:
    <a
      data-testid="budgets"
      href="/budgets"
      onClick={[Function]}
    >
      Budgets
    </a>
    null:
    <a
      data-testid="action"
      href="/"
      id="action"
      onClick={[Function]}
    >
      Actions
    </a>
  </div>
</div>
`;
